code/modules/vore/resizing/resize_vr.dm

```diff
@@ -64,28 +64,23 @@ var/const/RESIZE_A_SMALLTINY = (RESIZE_SMALL + RESIZE_TINY) / 2
  * Resizes the mob immediately to the desired mod, animating it growing/shrinking.
  * It can be used by anything that calls it.
  */
-/atom/movable/proc/in_dorms()
-	var/area/A = get_area(src)
-	return istype(A, /area/crew_quarters/sleep)
 
 /atom/movable/proc/size_range_check(size_select)		//both objects and mobs needs to have that
-	if((!in_dorms() && (size_select > 200 || size_select < 25)) || (size_select > 600 || size_select <1))
+	var/area/A = get_area(src) //Get the atom's area to check for size limit.
+	if((A.limit_mob_size && (size_select > 200 || size_select < 25)) || (size_select > 600 || size_select <1))
 		return FALSE
 	return TRUE
 
-/mob/living/proc/resize(var/new_size, var/animate = TRUE, var/mark_unnatural_size = TRUE)
+/mob/living/proc/resize(var/new_size, var/animate = TRUE, var/uncapped = FALSE)
+	if(!uncapped)
+		new_size = clamp(new_size, RESIZE_TINY, RESIZE_HUGE)
+		src.size_uncapped = FALSE
```

Comment:
What's the purpose of setting this false here?

Reply:
The reason I'm setting it here so that by default the function caps the size with that set to FALSE. This is to prevent majority of other code relying on this proc from breaking. Certain Medicines, NIFs as well as some other use cases relied on resize proc, which was expecting a third argument, which previously was used to determine if a Mob's unnaturally_resized variable should be set to TRUE or not. This also allows this same proc to be called in Admin verb without the capped size.